src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -170,7 +163,8 @@
   [rules :- Rules
    request :- ring/Request
    name :- schema/Str]
-  (if-let [ { matched-rule :rule matches :matches } (some #(match? % request) rules)]
+  (if-let [{matched-rule :rule matches :matches}
+           (some #(match? % request) (sort-rules rules))]
```

Comment:
Sorting the code "over here" only so it has an effect "over there" feels smelly too though.  I don't see an immediate need to modify the rules after transform-code either, but that might change at some point and we'd have to remember the temporal coupling here.

I'll make the change, but I don't think we should be so quick to assume performance implications.